archlinuxcn/lceda-pro/PKGBUILD

```diff
@@ -3,24 +3,24 @@
 
 pkgbase=lceda-pro
 pkgname=lceda-pro
-pkgver=2.1.42
+pkgver=2.1.44.1
 pkgrel=4
 pkgdesc="免费、专业、强大的国产PCB设计工具"
 arch=('x86_64' 'aarch64' 'loong64')
 url="https://pro.lceda.cn/"
-license=('custom')
+license=('LicenseRef-LCEDA-Proprietary')
 depends=(gtk3 nss alsa-lib electron libappindicator-gtk3 libnotify)
 install=${pkgname}.install
-source=("LICENSE"
+source=("LICENSE-$pkgver.html::https://lceda.cn/page/legal"
         "${pkgname}.install")
 source_x86_64=("${pkgname}-x86_64-${pkgver}.zip::https://image.lceda.cn/files/lceda-pro-linux-x64-${pkgver}.zip")
 source_aarch64=("${pkgname}-aarch64-${pkgver}.zip::https://image.lceda.cn/files/lceda-pro-linux-arm64-${pkgver}.zip")
 source_loong64=("${pkgname}-loong64-${pkgver}.zip::https://image.lceda.cn/files/lceda-pro-linux-loong64-${pkgver}.zip")
-sha256sums=('9b53bc19a98498c86019cc32a3ade6ad0ee4b12ba30686feb93132a5f0da52f5'
+sha256sums=('SKIP'
             'f8c3c7f65443801b8a70e40de7cdceade5dcd75974945695dd5a1bfb1f862e1a')
-sha256sums_x86_64=('4a6ded3805fda5bc78d426796d719bd598e96566d57e421223ef15391d97f65d')
-sha256sums_aarch64=('7a656db15f2ff60c697e611c8b7d8394d50805f7de3fe3b194867cfacdeddf1c')
-sha256sums_loong64=('35a8af2b0a5362ba7b70855406b5497c4cc64bc1f1ea2b88a0d1144e3876d940')
+sha256sums_x86_64=('3b5ea2ba2c450ec9fcaf035bb6c273694e399c3030ddb7f77eef85ed1cf617c4')
+sha256sums_aarch64=('8514fc3aead2221ee2a702bd2f88989b88617d55d999e5163d0b3bbac68e4fc5')
+sha256sums_loong64=('abf642c391649e06d5dfd6ce60d44e3f0bfdce16f63760cf26303b7f48a63dd6')
 
 package() {
     export LC_CTYPE="zh_CN.UTF-8"
```

Comment:
```
/startdir/PKGBUILD: line 26: warning: setlocale: LC_CTYPE: cannot change locale (zh_CN.UTF-8): No such file or directory
```
```
lceda-pro W: ELF file ('usr/lib/lceda-pro/resources/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node') lacks FULL RELRO, check LDFLAGS.
lceda-pro W: ELF file ('usr/lib/lceda-pro/resources/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node') lacks PIE.
lceda-pro W: ELF file ('usr/lib/lceda-pro/resources/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node') lacks GNU_PROPERTY_X86_FEATURE_1_SHSTK.
lceda-pro W: Directory (usr/lib/lceda-pro/resources/app/assets/chameleon/2.0.84.2739cbbe/html) is empty
lceda-pro W: Directory (usr/lib/lceda-pro/resources/app/assets/smt-ui/0.6.1.b2af29d2/assets/fonts) is empty
lceda-pro W: Unused shared library '/usr/lib/libpthread.so.0' by file ('usr/lib/lceda-pro/resources/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node')
```
```
PKGBUILD (lceda-pro) W: Reference to x86_64 should be changed to $CARCH
```

Reply:
> ```
> /startdir/PKGBUILD: line 26: warning: setlocale: LC_CTYPE: cannot change locale (zh_CN.UTF-8): No such file or directory
> ```

I think this line should be deleted but @taotieren marked it resolved in #2981

Reply:
> ```
> lceda-pro W: ELF file ('usr/lib/lceda-pro/resources/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node') lacks FULL RELRO, check LDFLAGS.
> lceda-pro W: ELF file ('usr/lib/lceda-pro/resources/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node') lacks PIE.
> lceda-pro W: ELF file ('usr/lib/lceda-pro/resources/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node') lacks GNU_PROPERTY_X86_FEATURE_1_SHSTK.
> lceda-pro W: Directory (usr/lib/lceda-pro/resources/app/assets/chameleon/2.0.84.2739cbbe/html) is empty
> lceda-pro W: Directory (usr/lib/lceda-pro/resources/app/assets/smt-ui/0.6.1.b2af29d2/assets/fonts) is empty
> lceda-pro W: Unused shared library '/usr/lib/libpthread.so.0' by file ('usr/lib/lceda-pro/resources/app/node_modules/sqlite3/lib/binding/napi-v3-linux-x64/node_sqlite3.node')
> ```

Since it is using binary source, I think we can't do much about it.

Reply:
> ```
> PKGBUILD (lceda-pro) W: Reference to x86_64 should be changed to $CARCH
> ```

I think it's a false positive. To make `updpkgsums` work on different architectures sources, I have to use x86_64 instead of $CARCH. I don't know if there is a better way to do it.

Reply:
> > ```
> > /startdir/PKGBUILD: line 26: warning: setlocale: LC_CTYPE: cannot change locale (zh_CN.UTF-8): No such file or directory
> > ```
> 
> I think this line should be deleted but @taotieren marked it resolved in #2981

You can consider deleting it and fix it later if there is any problem.